src/Time/Actions/ChangeStepSize.hpp

```diff
@@ -72,6 +72,14 @@ bool change_step_size(
     desired_step = -desired_step;
   }
 
+  if (abs(desired_step / current_step.slab().duration().value()) < 1.0e-10) {
+    ERROR(
+        "Chosen step is extremely small; this can indicate a flaw in the a "
```

Comment:
[No specific request.  Change something if you want.] The smallest representable time interval is `1/(2^31 - 1)` slabs, which in practice means `1/2^30`.

Reply:
It's a fair point that the steps can go far smaller than this error restricts, but I chose this as a practical limit so that the both simulation doesn't need to persist in a pathological state of extremely small step size for too long, but also legitimate step sizes weren't excluded. My rough estimate was that 1e-10 was a safe choice for 'almost certainly an unreasonable step size', but I could be convinced that this might be better set at a somewhat different value.

Reply:
Actually, 1e-10 is about an order of magnitude _smaller_ than 2^-30.

Reply:
Ah, right, thanks. I'll make the error check larger to be more meaningful.